dotCMS/src/main/java/com/dotcms/services/VanityUrlServices.java

```diff
@@ -0,0 +1,189 @@
+package com.dotcms.services;
+
+import com.dotcms.cache.VanityUrlCache;
+import com.dotcms.util.VanityUrlUtil;
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotcms.vanityurl.model.VanityUrl;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.IdentifierAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This service allows to invalidate the Vanity URL Cache
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 20, 2017.
+ */
+public class VanityUrlServices {
+
+    private static VanityUrlServices vanituUrlService;
+    private final VanityUrlCache vanityURLCache = CacheLocator.getVanityURLCache();
+    private final ContentletAPI contentletAPI = APILocator.getContentletAPI();
+    private final IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
+
+    /**
+     * Get the VanityUrlService singleton
+     *
+     * @return a VanityUrlServices
+     */
+    public static VanityUrlServices getInstance() {
+        if (vanituUrlService == null) {
+
+            synchronized (VanityUrlServices.class) {
+                vanituUrlService = new VanityUrlServices();
+            }
+
+        }
+        return vanituUrlService;
+    }
+
+    private VanityUrlServices() {
+
+    }
+
+    /**
+     * Remove the vanity URL from the vanityURLCache
+     *
+     * @param vanityUrl The vanity URL object
+     */
+    public void invalidateVanityUrl(VanityUrl vanityUrl) {
+        invalidateVanityUrl((Contentlet) vanityUrl);
+    }
+
+    /**
+     * Remove the vanity URL Contentlet from the vanityURLCache
+     *
+     * @param vanityUrl The vanity URL contentlet object
+     */
+    public void invalidateVanityUrl(Contentlet vanityUrl) {
+        invalidateAllVanityUrlVersions(vanityUrl);
+    }
+
+    /**
+     * Remove all the vanity Url contentlet versions from the vanity URL cache
+     *
+     * @param vanityUrl The vanity URL contentlet object
+     */
+    public void invalidateAllVanityUrlVersions(Contentlet vanityUrl) {
+        try {
+            Identifier identifier = identifierAPI.find(vanityUrl.getIdentifier());
+            List<Contentlet> contentletVersions = contentletAPI
+                    .findAllVersions(identifier, APILocator.systemUser(), false);
+            contentletVersions.stream()
+                    .filter(con -> vanityUrl.getLanguageId() == con.getLanguageId())
+                    .forEach((Contentlet con) -> removeFromCache(con));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)